sql-plugin/src/main/scala/org/apache/spark/sql/rapids/datetimeExpressions.scala

```diff
@@ -203,6 +203,70 @@ case class GpuTimeSub(start: Expression,
   }
 }
 
+case class GpuDateAddInterval(start: Expression,
+    interval: Expression,
+    timeZoneId: Option[String] = None)
+    extends GpuTimeMath(start, interval, timeZoneId) {
+
+  override def withTimeZone(timeZoneId: String): TimeZoneAwareExpression = {
+    copy(timeZoneId = Option(timeZoneId))
+  }
+
+  override def intervalMath(us_s: Scalar, us: ColumnVector): ColumnVector = {
+    us.add(us_s)
+  }
+
+  override def inputTypes: Seq[AbstractDataType] = Seq(DateType, CalendarIntervalType)
```

Comment:
inputTypes are wrong. It takes `Seq(DateType, TypeCollection(IntegerType, ShortType, ByteType))`

Reply:
DateAddInterval takes  `Seq[AbstractDataType] = Seq(DateType, CalendarIntervalType)` . Above comment on confusion between `DateAdd` and `DateAddInterval`.

Reply:
Yes, I apologize! :)

Reply:
No problem :)  Thanks for taking a look.